src/Models/ArtistsModel.php

```diff
@@ -0,0 +1,66 @@
+<?php
+require_once('src/Entities/Artist.php');
+class ArtistsModel
+{
+    private PDO $db;
+    public function __construct(PDO $db)
+    {
+        $this->db = $db;
+    }
+    public function getArtistsAlbumCount(int $artistId): Artist
```

Comment:
So I have no problem with you splitting this up into multiple queries - that's fine, and probably what I'd do vs trying to sandwich them all into one.

However, the issue I have is that you're using the artist entity multiple times here, and each time it is partially complete. More comments on specific methods to follow